devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -805,7 +805,7 @@ def zwaveEvent(UserCodeReport cmd) {
 				result << codeDeletedEvent(lockCodes, codeID)
 			} else {
 				map.value = "$codeID unset"
-				map.isStateChange = false
+				map.isStateChange = true
```

Comment:
During scanning if a slot is found to be empty it is not a good experience to display `unset` event as nothing has been `unset` actually. If there are lock codes in `lockCodes` attribute but during scanning a slot is found to be empty then DTH will generate `codeChanged` event with value `$codeID deleted`. SmartApps can use this event for updating their state.

Reply:
Perfect Rahul, that's a great solution.

Existing apps manage the users in the SmartApp, they keep track of the lock database through their own mechanisms, so they rely heavily on getting *every* notification from the lock to manage those databases. Without getting into the history of why SmartApps were designed this way, sometimes when the slot is supposed to be empty, they aren't! So there are safety mechanisms built in to handle clear codes, sometimes they *may* be empty and in those cases these notifications are critical for apps to track which slots were infact empty and which weren't.

Also how would you handle locks that don't report codes? How would the DTH reliably track codes without knowing which code was sent on the lock (you may be surprised but users do set codes on locks *while* using a SmartApp which leads to total havoc when locks don't report codes).